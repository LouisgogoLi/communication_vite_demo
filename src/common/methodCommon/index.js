import { thousandsSeparator } from './thousandsSeparator.js';
import { MousePosition } from './useMousePosition.js';
import { StateTodo } from './useState.js';

export const fnThousandsSeparator = thousandsSeparator;

export const fnMousePosition = MousePosition;
export const fnStateTodo = StateTodo;
