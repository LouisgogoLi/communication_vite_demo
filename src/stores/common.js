import { ref } from "vue";
import { defineStore, acceptHMRUpdate } from "pinia";

export const useCommonStore = defineStore("common", () => {
  const language = ref("en");

  return { language };
});

if (import.meta.hot) {
  import.meta.hot.accept(acceptHMRUpdate(useCommonStore, import.meta.hot));
}
