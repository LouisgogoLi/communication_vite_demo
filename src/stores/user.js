import { ref } from 'vue';
import { defineStore, acceptHMRUpdate } from 'pinia';

/**
 * Simulate a login
 * @param {string} a
 * @param {string} p
 */
function apiLogin(a, p) {
  if (a === 'ed' && p === 'ed') return Promise.resolve({ isAdmin: true });
  if (p === 'ed') return Promise.resolve({ isAdmin: false });
  return Promise.reject(new Error('invalid credentials'));
}

export const useUserStore = defineStore('user', () => {
  const name = ref('Eduardo');
  const isAdmin = ref(true);

  function logout() {
    this.$patch({
      name: '',
      isAdmin: false,
    });
    // we could do other stuff like redirecting the user
  }

  /**
   * Attempt to login a user
   * @param {string} user
   * @param {string} password
   */
  async function login(user, password) {
    const userData = await apiLogin(user, password);

    this.$patch({
      name: user,
      ...userData,
    });
  }

  return { name, isAdmin, logout, login };
});

if (import.meta.hot) {
  import.meta.hot.accept(acceptHMRUpdate(useUserStore, import.meta.hot));
}
