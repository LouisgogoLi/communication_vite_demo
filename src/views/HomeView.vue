<template>
  <div class="home">
    <img alt="Vue logo" src="@/assets/images/logo.png" />
    <HelloWorld msg="Welcome to Your Vue.js App" />
  </div>
</template>

<script>
export default {
  name: "HomeView",
};
</script>

<script setup>
// @ is an alias to /src
import HelloWorld from "@/components/HelloWorld.vue";
</script>
