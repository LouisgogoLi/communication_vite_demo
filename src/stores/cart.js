import { reactive, computed } from 'vue';
import { defineStore, acceptHMRUpdate } from 'pinia';
import { useUserStore } from './user';

export const useCartStore = defineStore('cart', () => {
  //cartRaw
  const rawItems = reactive([]);

  // @returns {Array<{ name: string; amount: number }>}
  const items = computed(() =>
    rawItems.reduce((items, item) => {
      const existingItem = items.find((it) => it.name === item);

      if (!existingItem) {
        items.push({ name: item, amount: 1 });
      } else {
        existingItem.amount++;
      }

      return items;
    }, [])
  );

  // Add item to the cart
  // @param {string} name
  function addItem(name) {
    rawItems.push(name);
  }

  //Remove item from the cart
  function removeItem(name) {
    const i = rawItems.lastIndexOf(name);
    if (i > -1) rawItems.splice(i, 1);
  }

  //購買後 清除items
  async function purchaseItems() {
    const user = useUserStore();
    if (!user.name) return;

    console.log('Purchasing', items.value);
    const n = items.value.length;
    return n;
  }

  return { rawItems, items, addItem, removeItem, purchaseItems };
});

if (import.meta.hot) {
  import.meta.hot.accept(acceptHMRUpdate(useCartStore, import.meta.hot));
}
