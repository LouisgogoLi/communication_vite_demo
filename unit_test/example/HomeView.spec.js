import { mount } from '@vue/test-utils';
import HomeView from '@/views/HomeView.vue';
import { createTestingPinia } from '@pinia/testing';
import { useCommonStore } from '@/stores/common';
import i18n from '@/i18n.js';

describe('HomeView.vue', () => {
  it('renders correctly', () => {
    const wrapper = mount(HomeView, {
      global: {
        plugins: [
          createTestingPinia({
            stubActions: false,
            plugins: [useCommonStore],
          }),
          i18n,
        ],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
