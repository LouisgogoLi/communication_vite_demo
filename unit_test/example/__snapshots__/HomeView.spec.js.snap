// Vitest Snapshot v1

exports[`HomeView.vue > renders correctly 1`] = `
"<select>
  <option value=\\"zh_tw\\">中文</option>
  <option value=\\"en\\">English</option>
  <option value=\\"ja\\">日本語</option>
</select>
<div class=\\"home\\"><img alt=\\"Vue logo\\" class=\\"logo\\" src=\\"/src/assets/svg/logo.svg\\" width=\\"125\\" height=\\"125\\">
  <div data-v-e17ea971=\\"\\">
    <h1 data-v-e17ea971=\\"\\">你好 i18n !!</h1>
    <p data-v-e17ea971=\\"\\">您已經成功創建了一個項目 <a href=\\"https://vitejs.dev/\\" target=\\"_blank\\" rel=\\"noopener\\" data-v-e17ea971=\\"\\">Vite</a> + <a href=\\"https://vuejs.org/\\" target=\\"_blank\\" rel=\\"noopener\\" data-v-e17ea971=\\"\\">Vue 3</a>. </p>
    <h3 data-v-e17ea971=\\"\\">文檔</h3>
    <ul data-v-e17ea971=\\"\\">
      <li data-v-e17ea971=\\"\\"><a target=\\"_blank\\" href=\\"https://vuejs.org/\\" data-v-e17ea971=\\"\\">Vue’s official documentation</a></li>
    </ul>
    <h3 data-v-e17ea971=\\"\\">配置</h3>
    <ul data-v-e17ea971=\\"\\">
      <li data-v-e17ea971=\\"\\"><a href=\\"https://vitejs.dev/guide/features.html\\" target=\\"_blank\\" data-v-e17ea971=\\"\\">Vite</a></li>
      <li data-v-e17ea971=\\"\\"><a href=\\"https://code.visualstudio.com/\\" target=\\"_blank\\" data-v-e17ea971=\\"\\">VSCode</a></li>
      <li data-v-e17ea971=\\"\\"><a href=\\"https://github.com/johnsoncodehk/volar\\" target=\\"_blank\\" data-v-e17ea971=\\"\\">Volar</a></li>
      <li data-v-e17ea971=\\"\\"><a href=\\"https://www.cypress.io/\\" target=\\"_blank\\" data-v-e17ea971=\\"\\">Cypress</a></li>
      <li data-v-e17ea971=\\"\\"><a href=\\"https://on.cypress.io/component\\" target=\\"_blank\\" data-v-e17ea971=\\"\\">Cypress Component Testing </a></li>
      <li data-v-e17ea971=\\"\\"><a href=\\"https://vitest.dev/\\" target=\\"_blank\\" data-v-e17ea971=\\"\\">Vitest</a></li>
    </ul>
    <h3 data-v-e17ea971=\\"\\">生態系統</h3>
    <ul data-v-e17ea971=\\"\\">
      <li data-v-e17ea971=\\"\\"><a href=\\"https://pinia.vuejs.org/\\" target=\\"_blank\\" data-v-e17ea971=\\"\\">Pinia</a></li>
      <li data-v-e17ea971=\\"\\"><a href=\\"https://router.vuejs.org/\\" target=\\"_blank\\" data-v-e17ea971=\\"\\">Vue Router</a></li>
      <li data-v-e17ea971=\\"\\"><a href=\\"https://test-utils.vuejs.org/\\" target=\\"_blank\\" data-v-e17ea971=\\"\\">Vue Test Utils</a></li>
      <li data-v-e17ea971=\\"\\"><a href=\\"https://github.com/vuejs/devtools\\" target=\\"_blank\\" data-v-e17ea971=\\"\\">Vue Dev Tools</a></li>
      <li data-v-e17ea971=\\"\\"><a href=\\"https://github.com/vuejs/awesome-vue\\" target=\\"_blank\\" data-v-e17ea971=\\"\\">Awesome Vue</a></li>
    </ul>
    <h3 data-v-e17ea971=\\"\\">社區</h3>
    <ul data-v-e17ea971=\\"\\">
      <li data-v-e17ea971=\\"\\"><a href=\\"https://chat.vuejs.org\\" target=\\"_blank\\" data-v-e17ea971=\\"\\">Vue Land</a></li>
      <li data-v-e17ea971=\\"\\"><a href=\\"https://stackoverflow.com/questions/tagged/vue.js\\" target=\\"_blank\\" data-v-e17ea971=\\"\\">StackOverflow</a></li>
      <li data-v-e17ea971=\\"\\"><a href=\\"https://twitter.com/vuejs\\" target=\\"_blank\\" data-v-e17ea971=\\"\\">@vuejs</a></li>
    </ul>
    <h3 data-v-e17ea971=\\"\\">協助 Vue</h3>
    <ul data-v-e17ea971=\\"\\">
      <li data-v-e17ea971=\\"\\"><a target=\\"_blank\\" href=\\"https://vuejs.org/sponsor/\\" data-v-e17ea971=\\"\\">becoming a sponsor</a></li>
    </ul>
  </div>
</div>"
`;
